test/end2end_test.go

```diff
@@ -2281,6 +2289,56 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestFailFastRPCRetryOnCredentialTimeout(t *testing.T) {
+	const minConnectTimeout = time.Millisecond
+	defer internal.TestingSetMinConnectTimeout(minConnectTimeout)()
+
+	before := grpc.DefaultBackoffConfig.BaseDelay
+	grpc.DefaultBackoffConfig.BaseDelay = 100 * time.Millisecond
+	defer func() {
+		grpc.DefaultBackoffConfig.BaseDelay = before
+	}()
+
+	te := newTest(t, env{name: "slow-tls", network: "tcp", security: "tls"})
```

Comment:
What's the downside to leaving this as it is? It more closely mirrors the true scenario we are interested in testing here. At this point, this issue is preventing CockroachDB from moving forward and producing a beta release, and the generally slow response times of the grpc-go team is not giving me confidence that such a rework of this PR will result in timely progress.